### PR TITLE
[vSphere][virtualmachine] Add support for additional metrics

### PR DESCRIPTION
**Pull request by lucian-ioan**
## Proposed commit message
Following beats code changes in https://github.com/elastic/beats/pull/40485, the following metrics need integration update:

Metrics | API | Field Mappings
-- | -- | --
VirtualMachine.Summary.QuickStats.UptimeSeconds | Summary | virtualmachine.uptime
VirtualMachine.Summary.OverallStatus | Summary | virtualmachine.status
VirtualMachine.Summary.Datastore| Summary | virtualmachine.datastore.names

Alerts |  
-- | -- 
triggerd_alarms.name | Name of the alarm.
triggerd_alarms.id | Unique key that identifies the alarm.
triggerd_alarms.status | Overall status of the alarm object.
triggerd_alarms.triggered_time | Time the alarm triggered.
triggerd_alarms.description | Description of the alarm.
triggerd_alarms.entity_name | alarm happened in respective entity

| Snapshot               |                      |
|-------------------------|------------------------------------------|
| snapshot.info.id         | Unique ID of the snapshot.               |
| snapshot.info.name       | Name of the snapshot.                    |
| snapshot.info.description| Description of the snapshot.             |
| snapshot.info.createtime | Time the snapshot was created.           |
| snapshot.info.state      | State of the snapshot (e.g., poweredOn). |
| snapshot.count           | Total number of snapshots.               |

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [x] Create Snapshot of Metricbeat changes.
- [x] Use Snapshot to make the changes in the vSphere Integration Host datastream.
- [x] Update the system and pipeline changes.
- [x] Add versioning to changelog.
- [x] Test the Integration for metric collection.
- [x] Test Integration from TSDB Perspective.
- [x] Code review.

## Related issues

- Closes https://github.com/elastic/obs-integration-team/issues/35

### CODE REVIEW

File: packages/vsphere/data_stream/virtualmachine/fields/fields.yml [hunk 1]
```diff
@@ -4,58 +4,89 @@
...
+        Used CPU in Mhz.
     - name: cpu.total.mhz
       type: long
-      metric_type: counter
-      description: Total CPU of virtualmachine in Mhz
+      description: >
+        Total CPU in Mhz.
     - name: cpu.free.mhz
       type: long
-      metric_type: gauge
```
Comment (lalit-satapathy) [inline packages/vsphere/data_stream/virtualmachine/fields/fields.yml:29]: Any reason to remove these metric_types?

File: packages/vsphere/data_stream/virtualmachine/fields/fields.yml [hunk 2]
```diff
@@ -4,58 +4,89 @@
...
+        Host id.
     - name: host.hostname
       type: keyword
-      description: Name of the host hosting the virtualmachine
+      description: >
+        Hostname of the host.
     - name: name
       type: keyword
-      # Reason to add as a dimension field: Each VM has Unique name assigned
-      dimension: true
```
Comment (ishleenk17) [inline packages/vsphere/data_stream/virtualmachine/fields/fields.yml:14]: @lucian-ioan : I suppose you are using the fields.yml directly from the beats code. 
That would not be correct as we don't consider tsdb there. 
Also, for any new fields please keep TSDB into consideration.

File: packages/vsphere/data_stream/virtualmachine/sample_event.json [hunk 3]
```diff
@@ -59,37 +59,58 @@
     },
     "vsphere": {
         "virtualmachine": {
```
Comment (ishleenk17) [inline packages/vsphere/data_stream/virtualmachine/sample_event.json:61]: Are these events autogenerated by system tests ?
Looks like these are being used from beats code ?

File: packages/vsphere/data_stream/virtualmachine/fields/fields.yml (hunk 1 above)
Comment (lucian-ioan) [reply packages/vsphere/data_stream/virtualmachine/fields/fields.yml:29]: No, reverting, tested first with beats code directly.

File: packages/vsphere/data_stream/virtualmachine/sample_event.json (hunk 3 above)
Comment (lucian-ioan) [reply packages/vsphere/data_stream/virtualmachine/sample_event.json:61]: Tested with beats code, regenerated it now.

File: packages/vsphere/changelog.yml [hunk 4]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.14.0-next"
```
Comment (ishleenk17) [inline packages/vsphere/changelog.yml:2]: Lets update this to latest 1.15.0-next

File: packages/vsphere/data_stream/virtualmachine/fields/fields.yml [hunk 5]
```diff
@@ -2,60 +2,140 @@
...
+              type: date
+              description: Snapshot creation time.
+            - name: state
+              type: keyword
+              description: Snapshot state (e.g., poweredOn).
+        - name: count
+          type: long
+          metric_type: gauge
+          description: The number of snapshots of this virtualmachine.
+    - name: triggerd_alarms.*
```
Comment (ishleenk17) [inline packages/vsphere/data_stream/virtualmachine/fields/fields.yml:137]: This needs to incorporate the latest change of this name change yesterday.
`"triggered_alamrs"`

File: packages/vsphere/data_stream/virtualmachine/manifest.yml [hunk 6]
```diff
@@ -13,6 +13,19 @@ streams:
         type: bool
         multi: false
         default: false
+      - name: period
+        type: text
+        title: Period
+        description: The vSphere performance API allows for collecting host and datastore metrics at various intervals, including real-time (every 20 seconds), 5 minutes, or longer durations. Properly configuring the "period" parameter is crucial to manage usage restrictions and ensure accurate data collection. For more information on optimizing your data collection setup, refer to the [Data Collection Intervals](https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.monitoring.doc/GUID-247646EA-A04B-411A-8DD4-62A3DCFCF49B.html) and [Data Collection Levels](https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.monitoring.doc/GUID-25800DE4-68E5-41CC-82D9-8811E27924BC.html) documentation.
```
Comment (ishleenk17) [inline packages/vsphere/data_stream/virtualmachine/manifest.yml:19]: Change the description as discussed yesterday

**Comment by niraj-elastic** [comment]
@lucian-ioan Please add tags in hbs file.

File: packages/vsphere/data_stream/virtualmachine/_dev/test/pipeline/test-virtualmachine.json-expected.json [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+                "ingested": "2024-09-02T09:44:17Z",
+                "module": "vsphere"
+            },
+            "host": {
+                "architecture": "x86_64",
+                "containerized": true,
+                "hostname": "elastic-agent-55444",
+                "id": "e744630f9d4f43dc818e497d221bd0b2",
+                "ip": [
+                    "172.18.0.4",
```
Comment (niraj-elastic) [inline packages/vsphere/data_stream/virtualmachine/_dev/test/pipeline/test-virtualmachine.json-expected.json:38]: Are this real IP's ? please change this if it belongs to any known VM.

**Review by niraj-elastic — CHANGES_REQUESTED** [review]

**Comment by ishleenk17** [comment]
Update description of commit PR with the snapshot fields as well

File: packages/vsphere/data_stream/virtualmachine/_dev/test/pipeline/test-virtualmachine.json-expected.json (hunk 7 above)
Comment (lucian-ioan) [reply packages/vsphere/data_stream/virtualmachine/_dev/test/pipeline/test-virtualmachine.json-expected.json:38]: Shouldn't be relevant anymore but I went ahead and modified all ips/domains.

**Review by ishleenk17 — APPROVED** [review]
Looks good!
Let the CI pass

**Pull request merged**